src/test/java/com/exasol/bucketfs/uploadnecessity/JsonRpcReadyWaitStrategy.java

```diff
@@ -0,0 +1,28 @@
+package com.exasol.bucketfs.uploadnecessity;
+
+/**
+ * Wait strategy that ensures the JSON RPC is available before it is used.
+ */
+public class JsonRpcReadyWaitStrategy {
+    boolean isReady = false;
```

Comment:
If you need this for testing: add a package-private getter to prevent modification.

```suggestion
    private boolean isReady = false;
```